email/provider_test.go

```diff
@@ -58,6 +59,32 @@ func testAccCheckEmailExists(n string) resource.TestCheckFunc {
 	}
 }
 
+func mockExponentialBackOff() (int, int, int) {
```

Comment:
This feels kind of redundant? Not sure how else we might want to go about this for now?

Reply:
From my understanding, a mock is a function that we can replace another function with, so we need to "inject" the mock into the program so that it can take place of another function.

For example, we can mock `smtp.SendMail` to return arbitrary responses by replacing that function with one that we control.

I've never done this in go but I feel like it should be the same in pretty much all languages. Here's a tutorial that looks promising:

https://quii.gitbook.io/learn-go-with-tests/go-fundamentals/mocking

Reply:
Specifically when they introduce `SpyCountdownOperations`. That part seems like what we want.

Reply:
hmm I thought about doing that, but I think we'll have to pair that terraform resource testing and then somehow verify exponential back off?

because if I mock the smtp.SendMail i'll have to have find a way to re-trigger the resourceCreate function (like we do in the first test) and then proceed to verify the delays, unless there is an easier way to manually trigger the resourceCreate and still verify behavior? (potentially by creating a another function with just the smtp.sendMail and retry and then mocking smtp.sendMail there?)